docs/dapp/sapphire/quickstart.md

```diff
@@ -0,0 +1,395 @@
+Let's make it happen!
+
+[higher level of security]: guide.md#writing-secure-dapps
+[dead person's switch]: https://en.wikipedia.org/wiki/Dead_man's_switch
+
+### Init a new Hardhat project
+
+1. Make & enter a new directory
+2. `npx hardhat` then create a TypeScript project.
+3. Install `@nomicfoundation/hardhat-toolbox` and its peer dependencies.
+
+### Add the Sapphire Testnet to Hardhat
+
+Open up your `hardhat.config.ts` and drop in these lines.
+They should remind you a lot about what happened with Truffle.
+
+```diff
+diff --git a/hardhat.config.ts b/hardhat.config.ts
+index 414e974..49c95f9 100644
+--- a/hardhat.config.ts
++++ b/hardhat.config.ts
+@@ -3,6 +3,15 @@ import "@nomicfoundation/hardhat-toolbox";
+
+ const config: HardhatUserConfig = {
+   solidity: "0.8.9",
++  networks: {
++    sapphire: {
++      chainId: 0x5aff,
++      url: "https://testnet.sapphire.oasis.dev",
++      accounts: [
++        process.env.PRIVATE_KEY ?? Buffer.alloc(0, 32).toString("hex"),
++      ],
++    },
++  },
+ };
+
+ export default config;
+```
+
+### Get the Contract
+
+This is a Sapphire tutorial and you're already a Solidity expert, so let's not
+bore you with explaining the gritty details of the contract.
+Start by pasting [Vigil.sol] into `contracts/Vigil.sol`.
+
+While you're there, also place [run-vigil.ts] into `scripts/run-vigil.ts`.
+We'll need that later.
+
+[Vigil.sol]: https://github.com/oasisprotocol/sapphire-paratime/blob/main/examples/hardhat/contracts/Vigil.sol
+[run-vigil.ts]: https://github.com/oasisprotocol/sapphire-paratime/blob/main/examples/hardhat/scripts/run-vigil.ts
+
+#### Vigil.sol, the interesting parts
+
+The key state variables are.
+
+```solidity
+    SecretMetadata[] public _metas;
+    bytes[] private _secrets;
+```
+
+* `_metas` is marked with `public` visibility, so despite the state itself being
+  encrypted and not readable directly, Solidity will generate a getter that will
+  do the decryption for you.
+* `_secrets` is `private` and therefore truly secret; only the contract can
+  access the data contained in this mapping.
+
+And the methods we'll care most about are
+
+* `createSecret`, which adds an entry to both `_metas` and `_secrets`.
+* `revealSecret`, which acts as an access-controlled getter for the data
+  contained with `_secrets`. Due to trusted execution and confidentiality, the
+  only way that the secret will get revealed is if execution proceeds all the
+  way to the end of the function and does not revert.
+
+The rest of the methods are useful if you actually intended to use this thing,
+but they demonstrate that developing for Sapphire is essentially the same as for
+Ethereum.
+You can even write tests against the Hardhat network 'n jazz.
+
+### Run the Contract
+
+And to wrap things up, we'll put `Vigil` through its paces.
+First, let's see what's actually going on.
+
+At the top of the file, we have our import du jour,
+[`@oasisprotocol/sapphire-paratime`].
```

Comment:
```suggestion
[@oasisprotocol/sapphire-paratime] which you can install by `npm i @oasisprotocol/sapphire-paratime --save-dev`.
```

Reply:
Again, this is for experienced solidity devs. If they can't figure out how to install deps, this is not for them.